codex/sales/reservations.nim

```diff
@@ -370,54 +398,61 @@ proc createAvailability*(
 
   return success(availability)
 
-proc createReservation*(
+method createReservation*(
   self: Reservations,
   availabilityId: AvailabilityId,
   slotSize: UInt256,
   requestId: RequestId,
   slotIndex: UInt256
-): Future[?!Reservation] {.async.} =
+): Future[?!Reservation] {.async, base.} =
 
-  trace "creating reservation", availabilityId, slotSize, requestId, slotIndex
+  withLock(self.availabilityLock):
+    without availabilityKey =? availabilityId.key, error:
+      return failure(error)
 
-  let reservation = Reservation.init(availabilityId, slotSize, requestId, slotIndex)
+    without availability =? await self.get(availabilityKey, Availability), error:
+      return failure(error)
 
-  without availabilityKey =? availabilityId.key, error:
-    return failure(error)
+    # We recheck that the found availability is still matching after we have lock acquired
```

Comment:
I think this is a bit misleading in the sense that it is only checked for size, and not "rechecked" for other qualifications (eg profitability). Plus this only makes sense if coming from `sales/states/pending`, which is currently the only callsite, but may be more in the future. Maybe more clear like, 
```suggestion
    # Check that the found availability has enough free space after the lock has been acquired, to prevent asynchronous Availiability modifications
```